targets/TARGET_STM/rtc_api.c

```diff
@@ -28,7 +28,7 @@
  *******************************************************************************
  */
 
-#if DEVICE_RTC
+#if DEVICE_RTC | DEVICE_LOWPOWERTIMER
```

Comment:
Not so easy...
If MBED_CONF_TARGET_LOWPOWERTIMER_LPTIM is enabled, RTC can be removed.